### PR TITLE
Fix jackson sbt assembly duplicate depends issue

### LINKED ISSUE #855: Sbt assembly  deduplicate error
i handle this  exception on **20.5.0**  at running `sbt assembly` after upgrade version 20.5.0

`
1 error was encountered during merge
[error] java.lang.RuntimeException: deduplicate: different file contents found in the following:
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.11.0/jackson-annotations-2.11.0.jar:module-info.class
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.11.0/jackson-core-2.11.0.jar:module-info.class
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.11.0/jackson-databind-2.11.0.jar:module-info.class
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/module/jackson-module-paranamer/2.11.0/jackson-module-paranamer-2.11.0.jar:module-info.class
[error]         at sbtassembly.Assembly$.applyStrategies(Assembly.scala:143)
[error]         at sbtassembly.Assembly$.x$1$lzycompute$1(Assembly.scala:25)
[error]         at sbtassembly.Assembly$.x$1$1(Assembly.scala:23)
[error]         at sbtassembly.Assembly$.stratMapping$lzycompute$1(Assembly.scala:23)
[error]         at sbtassembly.Assembly$.stratMapping$1(Assembly.scala:23)
[error]         at sbtassembly.Assembly$.inputs$lzycompute$1(Assembly.scala:68)
[error]         at sbtassembly.Assembly$.inputs$1(Assembly.scala:58)
[error]         at sbtassembly.Assembly$.apply(Assembly.scala:85)
[error]         at sbtassembly.Assembly$.$anonfun$assemblyTask$1(Assembly.scala:244)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error]         at sbt.Execute.work(Execute.scala:290)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error]         at java.base/java.lang.Thread.run(Thread.java:834)
[error] (assembly) deduplicate: different file contents found in the following:
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.11.0/jackson-annotations-2.11.0.jar:module-info.class
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.11.0/jackson-core-2.11.0.jar:module-info.class
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.11.0/jackson-databind-2.11.0.jar:module-info.class
[error] /Users/alikemal.ocalan/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/module/jackson-module-paranamer/2.11.0/jackson-module-paranamer-2.11.0.jar:module-info.class
`

### Expected behavior

As concisely as possible, describe the expected behavior.

### Actual behavior

As concisely as possible, describe the observed behavior.

### Steps to reproduce the behavior

Please list all relevant steps to reproduce the observed behavior.

### PR DESCRIPTION
**Pull request by alikemalocalan**
Problem
Fixes #855 

Explain the context and why you're making that change.  What is the
problem you're trying to solve? In some cases there is not a problem
and this can be thought of being the motivation for your change.

Solution

Describe the modifications you've done.

Result

What will change as a result of your pull request? Note that sometimes
this section is unnecessary because it is self-explanatory based on
the solution.

### CODE REVIEW

**Comment by ryanoneill** [comment]
Hi @alikemalocalan, thanks for the submission. Sorry that I missed it earlier this week. I'm catching up on the details now.

File: build.sbt [hunk 1]
```diff
@@ -497,8 +497,7 @@ lazy val finagleException = Project(
     name := "finagle-exception",
     libraryDependencies ++= Seq(
       util("codec")
-    ) ++ scroogeLibs,
-    libraryDependencies ++= jacksonLibs
+    ) ++ scroogeLibs
```
Comment (ryanoneill) [inline build.sbt:500]: finagle-exception uses jackson directly, so it really should have a dependency on jackson.

File: build.sbt [hunk 2]
```diff
@@ -520,7 +519,6 @@ lazy val finagleServersets = Project(
       ),
       "commons-lang" % "commons-lang" % "2.6"
     ),
-    libraryDependencies ++= jacksonLibs,
```
Comment (ryanoneill) [inline build.sbt:523]: finagle-serversets does too, so it should too.

File: build.sbt [hunk 3]
```diff
@@ -720,7 +716,7 @@ lazy val finagleMySQL = Project(
       util("stats"),
       caffeineLib,
       jsr305Lib
-    ) ++ jacksonLibs,
+    ),
```
Comment (ryanoneill) [inline build.sbt:719]: finagle-mysql too.

**Review by ryanoneill — COMMENTED** [review]
The other projects contain a dependency on jackson for perhaps not an obvious reason. For our open source projects like Finagle (which includes Finatra, Util, Scrooge, and TwitterServer), there are two build systems. One of which we use externally, sbt, while the other we use internally, pants -> eventually bazel. With that said, pants has stricter rules on the use of transitive dependencies, especially when types are used or exported from that dependency. So it's easiest for maintenance and management if our dependencies for our sbt projects match those closely with the ones for pants. That's why it might not be obvious why some of these projects contain a dependency on jackson.

I'm curious as to how changing things in this way fixes your issue.

**Comment by bryce-anderson** [comment]
@alikemalocalan, are you interested in addressing Ryan's feedback?

**Comment by alikemalocalan** [comment]
hi @bryce-anderson , i understood explanation of @ryanoneill , my solution isn't useful for like this project. Maybe next years , if you will  find more simple sbt sub-module solution, i can help you :D

**Pull request closed without merging**